mysql-test/include/gap_lock_error_update.inc

```diff
@@ -71,7 +71,6 @@ insert into u1 values (1,2,1,1) on duplicate key update c4=10;
 commit;
 begin;
 select * from u1 where c3=1 and c1 = 1 for update;
---error ER_UNKNOWN_ERROR
 select * from u1 where c3=1 for update;
```

Comment:
I don't believe so, the query should lock and return specific rows but not a true range of rows and gaps between rows. It should not be locking any space in between keys which would trigger the gap lock violation. So true, it is not a single row point select, but it is not a gap or range select either.

What I am curious about is how this behavior changed/test started failing without detection at the time of the cause. I _suspect_ something in one of the massive upstream merges got by without me noticing due to the noise of continuous mass failures of mtr cases. I'm tempted to start a history bisection to see if I can figure at what point this changed from my initial work on MYR-15 and later work in MYR-35. i.e. why did this pass (incorrectly) before and not now?